app/src/main/java/com/ataulm/whatsnext/account/SignInPresenter.kt

```diff
@@ -2,48 +2,34 @@ package com.ataulm.whatsnext.account
 
 import com.ataulm.whatsnext.Token
 import com.ataulm.whatsnext.WhatsNextService
-import io.reactivex.disposables.Disposable
-import kotlinx.coroutines.Dispatchers
-import kotlinx.coroutines.GlobalScope
-import kotlinx.coroutines.launch
-import kotlinx.coroutines.withContext
+import kotlinx.coroutines.*
 
 internal class SignInPresenter(
         private val service: WhatsNextService,
         private val screen: SignInScreen,
         private val callback: Callback
 ) {
 
-    private var disposable: Disposable? = null
-
-    fun startPresenting() {
-        screen.attach(onClickSignInCallback)
-    }
+    private val presenterScope = CoroutineScope(SupervisorJob() + Dispatchers.Main)
```

Comment:
While `ViewModel` has this behavior for us (see SearchViewModel below), we can still do it ourselves by defining a coroutine scope for all coroutines to run on in this presenter.

Each coroutine scope has a coroutine context which should have an associated job (`SupervisorJob()` in this case). Both `SupervisorJob()` and `Dispatchers.Main` are `CoroutineContexts` and these contexts can be combined (with the + operator) to yield another context.

We can keep a reference to the job separately so that we can cancel it when the presenter will be destroyed, but calling `presenterScope.cancel` will do this for us too.